File: app/src/main/java/com/labessence/biotech/finaldilution/compound/view/AddCompoundButtonView.java
package com.labessence.biotech.finaldilution.compound.view;

import android.content.Context;
import android.graphics.Canvas;
import android.graphics.Paint;
import android.graphics.Path;
import android.support.v4.content.ContextCompat;
import android.util.AttributeSet;

import com.labessence.biotech.finaldilution.R;

public class AddCompoundButtonView extends android.support.v7.widget.AppCompatTextView {
    public AddCompoundButtonView(Context context) {
        super(context);
    }

    public AddCompoundButtonView(Context context, AttributeSet attrs) {
        super(context, attrs);
    }

    public AddCompoundButtonView(Context context, AttributeSet attrs, int defStyleAttr) {
        super(context, attrs, defStyleAttr);
    }

    @Override
    protected void onDraw(Canvas canvas) {

        float yTop = 0;
        float yBottom = getHeight();
        float xLeft = 0;
        float xRight = getWidth();
        float xMid = getWidth() / 2.0f;
        float barWidth = getWidth() / 2.0f;

        Path path = new Path();

        // Top Direction Shape
        path.moveTo(xLeft, yBottom);
        path.moveTo(xMid - barWidth / 2, yBottom);
        path.cubicTo(xMid - barWidth / 4, yBottom, xMid - barWidth / 4, yTop, xMid, yTop);
        path.cubicTo(xMid + barWidth / 4, yTop, xMid + barWidth / 4, yBottom, xMid + barWidth / 2, yBottom);
        path.moveTo(xRight, yBottom);
        path.lineTo(xLeft, yBottom);

        path.close();

        Paint p = new Paint();
        p.setColor(ContextCompat.getColor(getContext(), R.color.grey1));
        p.setAntiAlias(true);

        canvas.drawPath(path, p);
        super.onDraw(canvas);

    }
}

File: app/src/main/java/com/labessence/biotech/finaldilution/compound/view/AddCompoundButtonView.kt
package com.labessence.biotech.finaldilution.compound.view

import android.content.Context
import android.graphics.Canvas
import android.graphics.Paint
import android.graphics.Path
import android.support.v4.content.ContextCompat
import android.util.AttributeSet

import com.labessence.biotech.finaldilution.R

class AddCompoundButtonView : android.support.v7.widget.AppCompatTextView {
    constructor(context: Context) : super(context) {}

    constructor(context: Context, attrs: AttributeSet) : super(context, attrs) {}

    constructor(context: Context, attrs: AttributeSet, defStyleAttr: Int) : super(context, attrs, defStyleAttr) {}

    override fun onDraw(canvas: Canvas) {

        val yTop = 0f
        val yBottom = height.toFloat()
        val xLeft = 0f
        val xRight = width.toFloat()
        val xMid = width / 2.0f
        val barWidth = width / 2.0f

        val path = Path()

        // Top Direction Shape
        path.moveTo(xLeft, yBottom)
        path.moveTo(xMid - barWidth / 2, yBottom)
        path.cubicTo(xMid - barWidth / 4, yBottom, xMid - barWidth / 4, yTop, xMid, yTop)
        path.cubicTo(xMid + barWidth / 4, yTop, xMid + barWidth / 4, yBottom, xMid + barWidth / 2, yBottom)
        path.moveTo(xRight, yBottom)
        path.lineTo(xLeft, yBottom)

        path.close()

        val p = Paint()
        p.color = ContextCompat.getColor(context, R.color.grey1)
        p.isAntiAlias = true

        canvas.drawPath(path, p)
        super.onDraw(canvas)

    }
}
